homeassistant/components/hive.py

```diff
@@ -0,0 +1,1000 @@
+"""Hive Integration - Platform."""
+import logging
+from datetime import datetime
+from homeassistant.helpers.discovery import load_platform
+
+REQUIREMENTS = ['pyhiveapi==0.0.41']
+
+HGO = None
+_LOGGER = logging.getLogger(__name__)
+DOMAIN = 'hive'
+
+
+class HivePlatformData:
+    """Initiate Hive PlatformData Class."""
+
+    min_max_data = {}
+
+
+class HiveSession:
+    """Initiate Hive Session Class."""
+
+    username = ""
+    password = ""
+    platform_data = HivePlatformData()
+    logging = False
+    hass = None
+    core = None
+    heating = None
+    hotwater = None
+    light = None
+    sensor = None
+    switch = None
+
+
+HSC = HiveSession()

```

Comment:
Rather than having a module-scope variable here, store the hive session in hass.data. There's one example here: https://github.com/home-assistant/home-assistant/blob/25cb7c652bf1a266829eea142196d1b9fdcec5a2/homeassistant/components/blink.py#L56

Reply:
Removed the module-scope variable